In [30]:
#Import library 
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#data importing

Sale=pd.read_excel('1 Sales Data.xlsx')

Sale.head(5)

Invoice Number       Date             Time   Zone Manager Customer  \
0        102234.0 2010-01-07  10:25:37.200000   East   Bruce     Polo   
1        102235.0 2010-01-07         12:00:00   East   Jacob      EEE   
2        102235.0 2010-01-07         14:19:12   East  Melvin    Marco   
3        102236.0 2010-01-07         12:57:00  South  Alicia      EEE   
4        102236.0 2010-01-07         14:51:36   East    Paul      Zee   

  Product   price  Units Sold    sales  
0     A001  485.0        42.0  20370.0  
1     B227  292.0        71.0  20732.0  
2     A009  544.0        87.0  47328.0  
3     D440  385.0        51.0  19635.0  
4      WP6  173.0        69.0  11937.0

### Check is there any null values in the data. Yes/ No

In [31]:
Sale.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26083 entries, 0 to 26082
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Invoice Number  26083 non-null  float64       
 1   Date            26083 non-null  datetime64[ns]
 2   Time            26083 non-null  object        
 3   Zone            26083 non-null  object        
 4   Manager         26083 non-null  object        
 5   Customer        26083 non-null  object        
 6   Product         26083 non-null  object        
 7   price           26083 non-null  float64       
 8   Units Sold      26083 non-null  float64       
 9   sales           26083 non-null  float64       
dtypes: datetime64[ns](1), float64(4), object(5)
memory usage: 2.0+ MB


### Find total sales.

In [32]:
a=Sale['sales'].sum()
a

1085806200.0

### Find total number of products sold.

In [33]:
b=Sale['Product '].value_counts().sum()
b

26083

### Find maximum Sales

In [34]:
c=Sale['sales'].max()
c

582000.0

### Find Minimum Sales

In [35]:
c=Sale['sales'].min()
c

865.0

### Find the Average Sales

In [36]:
d=Sale['sales'].mean()
d

41628.88471418165

### Find the Range of Sales

In [37]:
# Max- Min

f=582000.0 -865.0
f

581135.0

### Find the Standard Deviation of Sales

In [38]:
e=Sale['sales'].std()
e

30413.734244059717

### Find Average Sales by each zone

In [39]:
g=Sale.groupby('Zone').mean()['sales']
g

Zone
East     40856.468783
North    41933.083101
South    40869.279839
West     42309.618582
Name: sales, dtype: float64

### Find maximum sales by each zone.

In [40]:
g=Sale.groupby('Zone').max()['sales']
g

Zone
East     162656.0
North    165300.0
South    157760.0
West     582000.0
Name: sales, dtype: float64

### Find minimum sales by each zone.

In [41]:
g=Sale.groupby('Zone').min()['sales']
g

Zone
East     1015.0
North    1038.0
South     865.0
West      865.0
Name: sales, dtype: float64

### Find Manager Wise sales.


In [42]:
h=Sale.groupby('Manager').sum()['sales']
h

Manager
Aaron         26849758.0
Abbott        27098975.0
Abel          28249474.0
Alexandra     25981767.0
Alice         26573341.0
Alicia        25575993.0
Allan         90291223.0
Bruce         25533890.0
Grace         25789066.0
Hannah        27736138.0
Jack          27608183.0
Jacob         27505184.0
Jim          267417319.0
Lucy          24248866.0
Lydia         24951596.0
Malcolm       24938730.0
Melvin        29764672.0
Paul          28634113.0
Paxton        26878183.0
Rojer        116239848.0
Sarah         23662807.0
Tom          109184168.0
Tony          25092906.0
Name: sales, dtype: float64

### Find product wise sales.



In [43]:
h1=Sale.groupby('Product ').sum()['sales']
h1

Product 
A001    71165990.0
A002    83824255.0
A003    95344384.0
A009    80055584.0
A010    36526140.0
AL10    81071385.0
AL9     63099648.0
B227    43665096.0
B228    78313572.0
B229    59977816.0
D440    55816915.0
D441    50352855.0
G224    52050360.0
G225    73537764.0
KKK     60218836.0
WP6     25954152.0
WP7     74831448.0
Name: sales, dtype: float64

### Find zone wise product sales.


In [44]:
h2=Sale.groupby(['Zone','Product ']).sum()['sales']
h2

Zone  Product 
East  A001        17583190.0
      A002        26627745.0
      A003        23718990.0
      A009        24735680.0
      A010        11311020.0
                     ...    
West  G224        14399892.0
      G225        22970988.0
      KKK         16949557.0
      WP6          6044793.0
      WP7         24498808.0
Name: sales, Length: 68, dtype: float64

### Find Average units sold by each manager.


In [45]:
h3=Sale.groupby('Manager').mean()['Units Sold']
h3

Manager
Aaron        115.607595
Abbott       113.591772
Abel         117.488924
Alexandra    115.731013
Alice        111.643987
Alicia       114.828704
Allan        112.941204
Bruce        110.144817
Grace        115.830696
Hannah       116.110759
Jack         113.547468
Jacob        113.286585
Jim          112.900398
Lucy         110.412975
Lydia        115.243671
Malcolm      109.780063
Melvin       118.268589
Paul         119.487342
Paxton       111.237342
Rojer        112.920000
Sarah        110.454114
Tom          115.477492
Tony         108.530063
Name: Units Sold, dtype: float64

### Find average units sold by zone.


In [46]:
h4=Sale.groupby('Zone').mean()['Units Sold']
h4

Zone
East     112.711743
North    113.574048
South    112.649597
West     114.656522
Name: Units Sold, dtype: float64

### Find total sales by customer.


In [47]:
h5=Sale.groupby('Customer').sum()['sales']
h5

Customer
ABC      157258814.0
AX       205881436.0
EEE      204380240.0
Marco    258936887.0
Polo     153491179.0
Zee      105857644.0
Name: sales, dtype: float64

### Find total units sold by manager.


In [48]:
h6=Sale.groupby('Manager').sum()['Units Sold']
h6

Manager
Aaron         73064.0
Abbott        71790.0
Abel          74253.0
Alexandra     73142.0
Alice         70559.0
Alicia        74409.0
Allan        243953.0
Bruce         72255.0
Grace         73205.0
Hannah        73382.0
Jack          71762.0
Jacob         74316.0
Jim          737917.0
Lucy          69781.0
Lydia         72834.0
Malcolm       69381.0
Melvin        77939.0
Paul          75516.0
Paxton        70302.0
Rojer        316176.0
Sarah         69807.0
Tom          287308.0
Tony          68591.0
Name: Units Sold, dtype: float64

### Find Total Number of Orders

In [51]:
h7=Sale['Invoice Number'].value_counts().sum()
h7

26083

In [52]:
h7=Sale['Invoice Number'].count()
h7

26083

### Give an Inference from the data

# Conclusion


This is a Sale Data.The data contains 26083 rows and 10 columns

Check is there any null values in the data.  No

Find total sales.- 1085806200.0 

Find total number of products sold. - 26083

Find maximum Sales - 582000.0

Find Minimum Sales - 865.0

Find the Average Sales - 41628.88471418165

Find the Range of Sales - -581135.0

Find the Standard Deviation of Sales - 30413.734244059717

Find Average Sales by each zone 

 Zone

 East     40856.468783

 North    41933.083101

 South    40869.279839

 West     42309.618582

 Name: sales, dtype: float64

Find maximum sales by each zone.

 Zone

 East     162656.0

 North    165300.0

 South    157760.0

 West     582000.0

 Name: sales, dtype: float64


Find minimum sales by each zone.

Zone

East     1015.0

North    1038.0

South     865.0

West      865.0

Name: sales, dtype: float64
